mne/decoding/search_light.py

```diff
@@ -0,0 +1,332 @@
+# Author: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from .mixin import TransformerMixin
+from .base import BaseEstimator  # XXX why are these copied in weird places. We should just have an externals.sklearn, not manually copy/pasted objects  # noqa
+from ..parallel import parallel_func
+
+
+class SearchLight(BaseEstimator, TransformerMixin):
+    """Search Light
+    Fit, predict and score a series of models to each subset of the
+    dataset along the third dimension.
+
+    Parameters
+    ----------
+    base_estimator : object | None (default=None)
+        The base estimator to iteratively fit on a subset of the dataset. If
+        None, then the estimator is a Logistic Regression.
+    n_jobs : int, optional (default=1)
+        The number of jobs to run in parallel for both `fit` and `predict`.
+        If -1, then the number of jobs is set to the number of cores.
+    """
+    def __init__(self, base_estimator=None, n_jobs=1):
+        from sklearn.linear_model import LogisticRegression
+        self.base_estimator = (LogisticRegression() if base_estimator is None
+                               else base_estimator)
+        self.n_jobs = n_jobs
+        if not isinstance(self.n_jobs, int):
+            raise ValueError('n_jobs must be int, got %s' % n_jobs)
+
+    def fit_transform(self, X, y):
+        """
+        Fit and transform a series of independent estimators to the dataset.
+
+        Parameters
+        ----------
+        X : array, shape(n_samples, n_features, n_iteration)
+            The training input samples. For each iteration, a clone estimator
+            is fitted independently.
+        y : array, shape(n_samples,)
+            The target values.
+
+        Returns
+        -------
+        y_pred : array, shape(n_samples, n_iterations)
+            Predicted values for each estimator.
+        """
+        return self.fit(X, y).transform(X)
+
+    def fit(self, X, y):
+        """Fit a series of independent estimators to the dataset.
+
+        Parameters
+        ----------
+        X : array, shape(n_samples, n_features, n_iteration)
+            The training input samples. For each iteration, a clone estimator
+            is fitted independently.
+        y : array, shape(n_samples,)
+            The target values.
+
+        Returns
+        -------
+        self : object
+            Return self.
+        """
+        self.estimators_ = list()
+        parallel, p_func, n_jobs = parallel_func(_sl_fit, self.n_jobs)
+        estimators = parallel(
+            p_func(self.base_estimator, split, y)
+            for split in np.array_split(X, n_jobs, axis=2))
```

Comment:
> axis=-1 ?

@agramfort In fact, have X.shape > 3 complicates the API, because I need to store the X dimensions to ensure that the train and generalization have the same dims. Until we don't have a specific usecase I'll keep it simple.